packages/babel-plugin-relay/BabelPluginRelayMacro.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright (c) 2013-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @providesModule BabelPluginRelayMacro
+ * @flow
+ * @format
+ */
+
+'use strict';
+
+const {createMacro} = require('babel-macros');
+const compileGraphQLTag = require('./compileGraphQLTag');
+const getValidGraphQLTag = require('./getValidGraphQLTag');
+
+function BabelPluginRelayMacro({references, state, babel}) {
+  const {types: t} = babel;
+  Object.keys(references).forEach(referenceKey => {
+    references[referenceKey].forEach(reference => {
+      const path = reference.parentPath;
+      const ast = getValidGraphQLTag(path);
+      if (ast) {
+        compileGraphQLTag(t, path, state, ast);
+      }
+    });
+  });
+}
+
+module.exports = createMacro(BabelPluginRelayMacro);
```

Comment:
It looks like [`babel-plugin-relay`](https://facebook.github.io/relay/docs/babel-plugin-relay.html) is configurable. Part of the point of using the macro is so people don't need to update the babel configuration (so they can use it with create-react-app). However it is still handy to be able to configure specific plugins, so `babel-macros` has an experimental config feature. [Learn about config](https://github.com/kentcdodds/babel-macros/blob/master/other/docs/author.md#config-experimental).

Reply:
Thanks for the pointer. I didn't address this yet, as most of the options only pertain to Compat and Classic. Maybe @gaearon can help answer whether Compat and Classic should be supported?

Reply:
If configuration is an edge case, I'd suggest omitting it for now. That's something that can be added later and I think it's better to not slow down the merge-ability of this PR for something that not many folks will need/use.

Reply:
I don't think we would want to support classic and compat for anything new. The rest of the options are only to allow us to use the same plugin internally at FB and shouldn't be needed in OSS.